### ISSUE TITLE
recv crash when there is data to be received, apparent regression

### ISSUE THREAD

**Issue by ideka** [citra-emu/citra#1843]
I'm making a little client application for the 3DS. It communicates with a Windows server.

Any time the client calls recv, if the server sent data, the current Citra nightly (`HEAD-51ee2d2` as of this writing) crashes immediately on Windows 10 (the Linux version doesn't seem to have this issue?). This doesn't happen on a real New 3DS system.

I tried older versions of Citra, and found that the problem happes as early as version `HEAD-848effe` from the 5th of January.  Version `HEAD-ac15b3b` from the 4th of January doesn't have the problem. That's why I suspect this to be a regression.

**Comment by Subv** [comment]
For the record, ctrulib uses different soc functions depending on the size of the data to send/receive. We only implement one set of those functions, while the other is completely unimplemented, do you have a log of your application before it crashes?

**Comment by MerryMage** [comment]
`8191 == 0x2000 - 1`: https://github.com/smealum/ctrulib/blob/98fa2f7983d5a31dbc6c446a3478e90294243ebd/libctru/source/services/soc/soc_recvfrom.c#L124

**Comment by ideka** [comment]
Yep, yep. That's probably what the problem is. Thanks for the insight guys. [sendto, same deal](https://github.com/smealum/ctrulib/blob/98fa2f7983d5a31dbc6c446a3478e90294243ebd/libctru/source/services/soc/soc_sendto.c#L114).

When I try to `recv` more than that, Citra doesn't crash but the program behaves strangely, and a bunch of these are printed to the console:

```
[   2.236771] Service <Error> core\hle\service\service.cpp:Service::Interface::SyncRequest:68: unknown / unimplemented function 'recvfrom_other': port=soc:U, cmd_buff[1]=0x394, cmd_buff[2]=0xAAAAAAAA, cmd_buff[3]=0x0, cmd_buff[4]=0x0, cmd_buff[5]=0x20, cmd_buff[6]=0x2, cmd_buff[7]=0xAAAAAAAC, cmd_buff[8]=0x8
[   2.253485] HW.Memory <Error> core\memory.cpp:Memory::Write:130: unmapped Write32 0xAAAAAAAA @ 0x00000000
[   2.253511] HW.Memory <Error> core\memory.cpp:Memory::Write:130: unmapped Write32 0xAAAAAAAA @ 0x00000004
```

(I'm sending a bunch of 0xAA's btw.)

About the first problem, Citra prints the following before crashing:

```
[   0.199235] Loader <Info> core\loader\loader.cpp:Loader::LoadFile:134: Loading file rd3dsClient.3dsx as 3DSX...
[   0.201605] Service.FS <Error> core\file_sys\archive_romfs.cpp:FileSys::ArchiveFactory_RomFS::ArchiveFactory_RomFS:22: Unable to read RomFS!
[   0.229427] Service.APT <Warning> core\hle\service\apt\apt.cpp:Service::APT::GetLockHandle:111: (STUBBED) called handle=0x00018004 applet_attributes=0x00000000
[   0.229581] Service.APT <Warning> core\hle\service\apt\apt.cpp:Service::APT::Enable:119: (STUBBED) called attributes=0x00000000
[   0.229641] Service.AP
```

Last line cut off.

**Comment by MerryMage** [comment]
#1845 is related to this.

**Comment by verysimplyms** [comment]
Also having the exact same issue. My homebrew app works perfectly fine on an actual o3ds, but not on Citra.

**Comment by benderscruffy** [comment]
Has this been fixed ??
Please update
